### PR TITLE
CAMEL-15356: Add Azure EventHubs component

### PR DESCRIPTION
**Pull request by omarsmak**
This adds Azure EventHubs component. I am using here the producer async, hence I'd appreciate the feedback here if I used the callbacks in the correct way cc @davsclaus

### CODE REVIEW

File: components/camel-azure-eventhubs/src/main/java/org/apache/camel/component/azure/eventhubs/operations/EventHubsProducerOperations.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        ObjectHelper.notNull(callback, "callback cannot be null");
+
+        final SendOptions sendOptions = createSendOptions(configurationOptionsProxy.getPartitionKey(exchange), configurationOptionsProxy.getPartitionId(exchange));
+        final Iterable<EventData> eventData = createEventData(exchange);
+
+        return sendAsyncEvents(eventData, sendOptions, exchange, callback);
+    }
+
+    private boolean sendAsyncEvents(final Iterable<EventData> eventData, final SendOptions sendOptions, final Exchange exchange, final AsyncCallback asyncCallback) {
+        final AtomicBoolean done = new AtomicBoolean(false);
```
Comment (davsclaus) [inline components/camel-azure-eventhubs/src/main/java/org/apache/camel/component/azure/eventhubs/operations/EventHubsProducerOperations.java:59]: The asyncCallback must only be invoked once, not sure if the 3 branches below can cause it to be invoked twice etc.

So you should essentially only return true if the method exits early and its the same thread that called this method that is calling the callback.

So I would remove the atomic boolean, and make this method return false. And then in the sendAsyncEventsWithSuitableMethod you call the callback with (false) as parameter as it must match what this method returned.

Comment (davsclaus) [reply components/camel-azure-eventhubs/src/main/java/org/apache/camel/component/azure/eventhubs/operations/EventHubsProducerOperations.java:59]: Also does the subcribe event below make any sense to react on? I would assume its either only the error or the completion.

**Comment by omarsmak** [comment]
@davsclaus thanks for the comments. Yeah, that subscribe event is not needed as it invokes on every emitted event which is not needed here, hence I removed it here. 
I *guess* now the callbacks the should be correct plus the other comments you mentioned, please take a look and let me know if anything further needed

File: components/camel-azure-eventhubs/src/main/java/org/apache/camel/component/azure/eventhubs/operations/EventHubsProducerOperations.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+        if (ObjectHelper.isEmpty(sendOptions)) {
+            deferredSendResult = producerAsyncClient.send(eventData);
+        } else {
+            deferredSendResult = producerAsyncClient.send(eventData, sendOptions);
+        }
+
+        // call callback on false since this process is running on the scheduler method of reactor and not needed
+        // to have callback to returns true. Also false, will tell camel async to watch method to which it returns
+        asyncCallback.done(false);
```
Comment (davsclaus) [inline components/camel-azure-eventhubs/src/main/java/org/apache/camel/component/azure/eventhubs/operations/EventHubsProducerOperations.java:83]: I would asumme this asyncCallback.done(false) should be in the previous method inside those 2 callbacks after ( // error but we continue) and also after ( // we are done from everything, so mark it as sync done). As this Camel callback should only be invoked when the azure client is fully complete with its work (either in error or with success).

**Comment by omarsmak** [comment]
okay I think it *should* be correct now

**Comment by omarsmak** [comment]
SpringBoot starter will follow up soon

**Pull request merged**